cmd/goal/clerk.go

```diff
@@ -1070,7 +1096,10 @@ var compileCmd = &cobra.Command{
 				}
 			}
 			shouldPrintAdditionalInfo := outname != stdoutFilenameValue
-			program, sourceMap := assembleFileWithMap(fname, true)
+			program, sourceMap, err := assembleFileWithMap(fname, outname, true)
+			if err != nil {
+				reportErrorf("Could not assemble: %s", err)
```

Comment:
Usually, we use %v in `reportErrorf` to print an error.  I'm not sure what this will do, since `err` won't have `String()` method.

Reply:
ehh, Go Playground says it's fine.

Reply:
I did not know, that Printf has a rule for Errors (that is applied before the rule for .String())
4. If an operand implements the error interface, the Error method will be invoked to convert the object to a string, which will then be formatted as required by the verb (if any).